Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -218,6 +284,49 @@ private class LocalVariableDeclarationVisitor : SyntaxTreeVisitor {
       public override void Visit(LocalVariable localVariable) {
         block.Symbols.Add(new VariableSymbol(block, localVariable));
       }
+
+      public override void Visit(LambdaExpr lambdaExpression) {
+        ProcessBoundVariableBearingExpression(lambdaExpression, () => base.Visit(lambdaExpression));
+      }
+
+      public override void Visit(ForallExpr forallExpression) {
+        ProcessBoundVariableBearingExpression(forallExpression, () => base.Visit(forallExpression));
+      }
+
+      public override void Visit(ExistsExpr existExpression) {
+        ProcessBoundVariableBearingExpression(existExpression, () => base.Visit(existExpression));
+      }
+
+      public override void Visit(SetComprehension setComprehension) {
+        ProcessBoundVariableBearingExpression(setComprehension, () => base.Visit(setComprehension));
+      }
+
+      public override void Visit(MapComprehension mapComprehension) {
+        ProcessBoundVariableBearingExpression(mapComprehension, () => base.Visit(mapComprehension));
+      }
+
+      public override void Visit(LetExpr letExpression) {
+        ProcessBoundVariableBearingExpression(letExpression, () => base.Visit(letExpression));
+      }
+
+      private void ProcessBoundVariableBearingExpression<TExpr>(
```

Comment:
Is the type parameter necessary?

Reply:
It was when, instead of providing a closure for the base.Visit, I was calling base.Visit() inside ProcessBoundVariableBearingExpression(). however, the dynamic dispatch led to base.Visit(Expression) which triggered an infinite loop, so I resignated to using the closure, and yes, the type parameter is indeed no longer necessary.